app/javascript/packages/personal-key-input/index.js

```diff
@@ -0,0 +1,22 @@
+import base32Decode from 'base32-decode';
+import base32Encode from 'base32-encode';
+
+/**
+ * Coerce mistaken user input from 'problem' letters:
+ * https://en.wikipedia.org/wiki/Base32#Crockford.27s_Base32
+ *
+ * @param {string} value User-provided text input
+ *
+ * @return {string} Encoded input
+ */
+export function encodeInput(value) {
+  value = value.replace(/-/g, '');
+  value = base32Decode(value, 'Crockford');
+  value = base32Encode(value, 'Crockford');
+
+  // Add back the dashes
+  value = value.toString().match(/.{4}/g).join('-');
```

Comment:
I think the TypeScript error here is legit:

```
app/javascript/packages/personal-key-input/index.js:18:11 - error TS2531: Object is possibly 'null'.
18   value = value.toString().match(/.{4}/g).join('-');
```

If someone passes a string which isn't long enough, an error can occur:

```js
encodeInput('abc');
```

```
  value = value.toString().match(/.{4}/g).join('-');
                                         ^
TypeError: Cannot read property 'join' of null
    at encodeInput (file:///Users/andrewmduthie/Documents/Code/identity-idp/app/javascript/packages/personal-key-input/index.js:17:42)
    at file:///Users/andrewmduthie/Documents/Code/identity-idp/app/javascript/packages/personal-key-input/index.js:23:1
    at ModuleJob.run (internal/modules/esm/module_job.js:146:23)
    at async Loader.import (internal/modules/esm/loader.js:165:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)
```

Maybe a fallback is to create an array of the value itself if there's no `match`?

```suggestion
  value = (value.match(/.{4}/g) || [value]).join('-');
```

Also, I think the `toString` is unnecessary if we can rely on `base32Encode` to give us a string.